> can you post the log of the qt build failure when using `-pch`?

```
$ make -C depends qt
...
Running configuration tests...
...
Checking for precompiled header support... 
+ cd /home/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.9.8-991caec5731/qtbase/config.tests/common/pch && PKG_CONFIG_SYSROOT_DIR=/ PKG_CONFIG_LIBDIR=/home/hebasto/bitcoin/depends/x86_64-pc-linux-gnu/lib/pkgconfig /home/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.9.8-991caec5731/qtbase/bin/qmake "CONFIG -= qt debug_and_release app_bundle lib_bundle" "CONFIG += static warn_off console single_arch" -early "CONFIG += cross_compile" /home/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.9.8-991caec5731/qtbase/config.tests/common/pch
+ cd /home/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.9.8-991caec5731/qtbase/config.tests/common/pch && MAKEFLAGS= /usr/bin/gmake
> gmake[1]: Entering directory '/home/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.9.8-991caec5731/qtbase/config.tests/common/pch'
> g++ -m64 -pipe -pipe -O2 -I/home/hebasto/bitcoin/depends/x86_64-pc-linux-gnu/include -O2 -w -fPIC  -I. -I/home/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.9.8-991caec5731/qtbase/mkspecs/linux-g++-64 -x c++-header -c header.h -o .pch/pch.gch/c++
> g++ -c -include .pch/pch -m64 -pipe -pipe -O2 -I/home/hebasto/bitcoin/depends/x86_64-pc-linux-gnu/include -O2 -w -fPIC  -I. -I/home/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.9.8-991caec5731/qtbase/mkspecs/linux-g++-64 -o source.o source.cpp
> cc1plus: error: one or more PCH files were found, but they were invalid
> cc1plus: error: use -Winvalid-pch for more information
> cc1plus: fatal error: .pch/pch: No such file or directory
> compilation terminated.
> gmake[1]: *** [Makefile:183: source.o] Error 1
> gmake[1]: Leaving directory '/home/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.9.8-991caec5731/qtbase/config.tests/common/pch'
test config.qtbase.tests.precompile_header FAILED
...
ERROR: Feature 'precompile_header' was enabled, but the pre-condition 'config.msvc || tests.precompile_header' failed.
make: *** [funcs.mk:268: /home/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.9.8-991caec5731/qtbase/.stamp_configured] Error 3
make: Leaving directory '/home/hebasto/bitcoin/depends'
```

--- a/ci/test/00_setup_env_i686_centos.sh
+++ b/ci/test/00_setup_env_i686_centos.sh
@@ -11,7 +11,6 @@ export CONTAINER_NAME=ci_i686_centos_8
 export DOCKER_NAME_TAG=centos:8
 export DOCKER_PACKAGES="gcc-c++ glibc-devel.x86_64 libstdc++-devel.x86_64 glibc-devel.i686 libstdc++-devel.i686 ccache libtool make git python3 python3-zmq which patch lbzip2 dash rsync coreutils"
 export GOAL="install"
-export DEP_OPTS="NO_QT=1"  # Gui disabled for now to avoid build failures
-export BITCOIN_CONFIG="--enable-zmq --with-gui=no --enable-reduce-exports --with-boost-process"
+export BITCOIN_CONFIG="--enable-zmq --with-gui=qt5 --enable-reduce-exports --with-boost-process"
 export CONFIG_SHELL="/bin/dash"
 export TEST_RUNNER_ENV="LC_ALL=en_US.UTF-8"

--- a/depends/packages/qt.mk
+++ b/depends/packages/qt.mk
@@ -69,7 +69,6 @@ $(package)_config_opts += -nomake examples
 $(package)_config_opts += -nomake tests
 $(package)_config_opts += -opensource
 $(package)_config_opts += -optimized-tools
-$(package)_config_opts += -pch
 $(package)_config_opts += -pkg-config
 $(package)_config_opts += -prefix $(host_prefix)
 $(package)_config_opts += -qt-libpng
@@ -117,6 +116,7 @@ $(package)_config_opts += -no-feature-xml
 
 $(package)_config_opts_darwin = -no-dbus
 $(package)_config_opts_darwin += -no-opengl
+$(package)_config_opts_darwin += -pch
 
 ifneq ($(build_os),darwin)
 $(package)_config_opts_darwin += -xplatform macx-clang-linux
@@ -149,6 +149,7 @@ $(package)_config_opts_mingw32 = -no-opengl
 $(package)_config_opts_mingw32 += -no-dbus
 $(package)_config_opts_mingw32 += -xplatform win32-g++
 $(package)_config_opts_mingw32 += -device-option CROSS_COMPILE="$(host)-"
+$(package)_config_opts_mingw32 += -pch
 
 $(package)_config_opts_android = -xplatform android-clang
 $(package)_config_opts_android += -android-sdk $(ANDROID_SDK)
@@ -164,6 +165,7 @@ $(package)_config_opts_android += -qt-freetype
 $(package)_config_opts_android += -no-fontconfig
 $(package)_config_opts_android += -L $(host_prefix)/lib
 $(package)_config_opts_android += -I $(host_prefix)/include
+$(package)_config_opts_android += -pch
 
 $(package)_config_opts_aarch64_android += -android-arch arm64-v8a
 $(package)_config_opts_armv7a_android += -android-arch armeabi-v7a

